File: src/main/java/org/itmo/isLab1/dragons/dto/DragonCreateDto.java
package org.itmo.isLab1.dragons.dto;


import jakarta.validation.constraints.*;
import lombok.Data;

import org.itmo.isLab1.dragons.enums.DragonType;
import org.itmo.isLab1.dragons.enums.DragonCharacter;
import org.itmo.isLab1.people.enums.Color;


@Data
public class DragonCreateDto {
    @NotNull
    @NotBlank
    private String name;

    @NotNull
    private Integer coordinatesId;

    private Integer caveId;

    private Integer killerId;

    @Min(0)
    private Integer age;

    private Color сolor;
  
    private DragonType type;

    @NotNull
    private DragonCharacter character;

    private Integer headId;
}

File: frontend/src/interfaces/dto/dragons/DragonCreateDto.ts
import { DragonType } from '../../models/DragonType';
import { Color } from '../../models/Colors';
import { DragonCharacter } from '../../models/DragonCharacter';

export interface DragonCreateDto {
    name: string;
    coordinatesId: number;
    caveId?: number;
    killerId?: number;
    age?: number;
    color?: Color;
    type?: DragonType;
    character: DragonCharacter;
    headId?: number;
}